asmcomp/selectgen.ml

```diff
@@ -211,6 +211,8 @@ let join_array env rs =
       done;
       Some res
 
+let contains_calls = ref false
+
```

Comment:
A mutable boolean instance value wouldn't work because the traversal uses functional object copies (`{< ... >}`). Making the reference an instance variable (`val contains_call = ref false`) would still be a good idea, though; this way, it is implicitly reset in `Selection.findecl` when the new class instance is created.

Reply:
(In that case please include a comment on the why it's a per-instance reference instead of an instance mutable variable, it's a bit tricky.)